js/ui/map.js

```diff
@@ -287,6 +287,27 @@ util.extend(Map.prototype, /** @lends Map.prototype */{
     },
 
     /**
+     * Set constraint on the map's geographical bounds.
+     *
+     * @param {LngLatBounds | Array<Array<number>>} lnglatbounds optional Desired max bounds of the map. If ommitted, function removes any bounds constraints on the map.
```

Comment:
Instead of marking the parameter as optional, let's specify the type as `LngLatBounds | Array<Array<number>> | undefined` and encourage calling it as `setMaxBounds(undefined)`. The implementation stays the same, but `undefined` is a more explicit way of saying "unset the max bounds" than omitting the parameter.